lib/states/pokemon_list_state.dart

```diff
@@ -3,16 +3,33 @@ part of 'states.dart';
 class PokemonListState with ChangeNotifier {
   static const int _itemsPerPage = 20;
 
-  PokemonListState(this._getPokemonsUseCase);
+  PokemonListState(this._getPokemonsUseCase, this._searchPokemonUseCase);
 
   final GetPokemonsUseCase _getPokemonsUseCase;
+  final SearchPokemonUseCase _searchPokemonUseCase;
 
   int page = 1;
   bool loading = true;
   bool canLoadMore = true;
   bool isError = false;
   List<Pokemon> pokemons = [];
 
+  void searchPokemon(String pokemonName) async {
+    isError = false;
+    loading = true;
+    notifyListeners();
+
+    try {
+      pokemons = await _searchPokemonUseCase(
+          SearchPokemonParams(pokemonName: pokemonName));
```

Comment:
```suggestion
          SearchPokemonParams(pokemonName: pokemonName),);
```
Please remember to add the little `,` to make the formatter working nicely